src/Agent.Worker/Build/TrackingConfig.cs

```diff
@@ -34,11 +34,11 @@ public TrackingConfig()
 
             // Set the other properties.
             CollectionId = copy.CollectionId;
+            CollectionUrl = executionContext.Variables.System_TFCollectionUrl;
             DefinitionId = copy.DefinitionId;
             HashKey = copy.HashKey;
             RepositoryUrl = copy.RepositoryUrl;
             System = copy.System;
-            SetCollectionName(executionContext);
```

Comment:
Collection in the domain makes parsing out the CollectionName unreliable. This was only required as an info field anyway, so doesn't really matter. As a replacement, I added the CollectionUrl instead as the info field.